storybookjs/storybook
# CLI: Add `--build-only` option to `start-storybook` (useful to warm up the cache)

### Issues:

Related to Discussion #13737 (where I initially described the use-case)
Related to #13266
Related to #13789
Related to #13784

---

I’d like a deterministic way to warm up a reusable cache for `start-storybook` without needing to start the server (nor opening the browser), which ideally would ensure that the next time somebody runs `start-storybook`, the cache is picked up and it runs faster.

For example, in the proprietary equivalent to GitHub Codespaces we run internally at Shopify, a warm cache means reducing Storybook's boot time by 30s (from 1m to 30s).

This is close to `--ci` but without the server. It's also close to `--smoke-test`, but without outputting stats, and with properly working caches.

## What I did

I added a new CLI option that intends to run everything a standard `start-storybook` would, but doesn't start the server.

## How to test

Assuming `yarn bootstrap` ran successfully,

1. `cd examples/cra-ts-examples`
2. `rm -rf node_modules/.cache/storybook`
2. `yarn start-storybook --build-only`: should exit without opening a browser
3. `yarn start-storybook`: should run faster

![Screen_Shot_2021-02-02_at_6_12_11_PM](https://user-images.githubusercontent.com/85783/106688369-d3e3af00-6582-11eb-9819-7d9f761e1a65.png)
